lib/config-validator.ts

```diff
@@ -50,6 +50,26 @@ type PackageJson = {
 };
 
 (async () => {
+  for (const file of process.argv.slice(2)) {
+    try {
+      if (!(await pathExists(file))) {
+        returnVal = 1;
+        logger.error(`${file} does not exist`);
+        break;
+      }
+      const parsedContent = await getParsedContent(file);
+      try {
+        logger.info(`Validating ${file}`);
+        await validate(file, parsedContent);
+      } catch (err) {
+        logger.warn({ err }, `${file} is not valid Renovate config`);
+        returnVal = 1;
+      }
+    } catch (err) {
+      logger.warn({ err }, `${file} could not be parsed`);
+      returnVal = 1;
+    }
+  }
   for (const file of configFileNames.filter(
```

Comment:
correct me if i'm wrong but 
let's say you entered "renovate.json", it will pass through the first loop, 
then will continue to the other loop, where it will check also for "renovate.json" in current repository and validate twice
if the file exists here.

if you want to set your Own files, or list of files, maybe override the actual configFileNames and check only the files that you entered. hence deleting the extra code you added which as far as i can see is almost the same as below

Reply:
Yes, this is how I did it in my initial commit, but I was asked to handle it slightly differently as you noticed

Reply:
i dont think this is what he meant though, we just need to check if files exist and then add them to the config file names,

there's a check already if they exist or not and we dont FAIL on it we just continue, : 
https://github.com/renovatebot/renovate/blob/ef470f7f923a4c03e4c6204094dd508f0a4eef96/lib/config-validator.ts#L57-L59

what you did before
```
    ...configFileNames.filter((name) => name !== 'package.json'),
    ...process.argv.slice(2),
```

looks good to me, if it causes any problem in your tests for any file that doesn't exist, then you better check if files exist and remove them from array before you merge them with the `configFileNames`.

i dont agree on copying the whole functionality just to add 1 file when we already have the same handling

Reply:
![image](https://user-images.githubusercontent.com/42116482/178219078-c609c1d1-847c-48ec-b30a-5b624e88d48e.png)

Reply:
my intention is to only validate cli args passed file names when passed and fall back to current behavior if no file is passed.

Reply:
Alright, I've wrapped it into an if statement